src/client/components/Reservations/ReservationForm.jsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState, useEffect } from "react";
+import "./Reservation.css";
+
+function ReservationForm({ mealId }) {
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [phone, setPhone] = useState("");
+  const [numberOfGuests, setNumberOfGuests] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [availableReservations, setAvailableReservations] = useState(0);
+
+  const handleFormSubmit = () => {
+    const bookingData = {
+      meal_id: mealId,
+      contact_name: name,
+      contact_email: email,
+      contact_phonenumber: phone,
+      number_of_guests: Number(numberOfGuests),
+      created_date: new Date().toJSON().slice(0, 10),
+    };
+    setLoading(true);
+    fetch("/api/reservations", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(bookingData),
+    })
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error("Something went wrong.");
+        }
+        return response.json();
+      })
+      .then((data) => {
+        console.log(data);
+        setName("");
+        setEmail("");
+        setPhone("");
+        setNumberOfGuests("");
+        setLoading(false);
```

Comment:
This could be an object as a state, so it becomes a bit more structured and easier to manage 🙏🏽

Reply:
Thank you @shpomp for your feedback! :)